pkg/controller/service/service.go

```diff
@@ -18,18 +18,21 @@ func RenderServices(req router.Request, resp router.Response) error {
 		return err
 	}
 	resp.Objects(objs...)
+	svcInstance.Status.HasService = len(objs) > 0
 
 	objs, err = publish.ServiceLoadBalancer(req, svcInstance)
 	if err != nil {
 		return err
 	}
 	resp.Objects(objs...)
+	svcInstance.Status.HasService = svcInstance.Status.HasService || len(objs) > 0
 
 	objs, err = publish.Ingress(req, svcInstance)
 	if err != nil {
 		return err
 	}
 	resp.Objects(objs...)
 
+	resp.Objects(svcInstance)
```

Comment:
This seems weird but do you want to put svcInstance into it's own `resp.Objects`? I think it should automatically update the status if you set it

Reply:
You're correct that this isn't necessary for running Acorn. However, this allows the unit tests to ensure that fields are set correctly after the handler runs.

Reply:
Hmm ok. It would be weird if we are just adding this for sake of unit test, can the test be asserting the original object instead of the object in resp.Objects?

Reply:
That wouldn't be possible with the testing framework we use for these handlers.

Functionally, it doesn't change anything: `baaah` will pull it out and update its status no matter what. This is similar to the way `DeploySpec` works. The `AppInstance` gets add to `resp.Objects` when we set the conditon.